modules/extreme_year.py

```diff
@@ -0,0 +1,164 @@
+          an int: either the lowest temperature or -1000 if there is no entry
+    """
+    return int(line[3]) if line[3] != "" else -1000
+
+
+def get_max_humidity(line):
+    """
+    Returns the maximum humidity from the line read of a weather file
+    Args:
+        line(list): a list of strings containing different fields at different index
+                    please have a look at any weatherfile for more clarity
+    Returns:
+          an int: either the max humidity or -1000 if there is no entry
+    """
+    return int(line[7]) if line[7] != "" else -1000
+
+
+def calculate_extremes(year, path):
+    """
+    This function reads all the files at the given path which contain the given pattern
+    from each file that matches the pattern finds the following:
+    1. max highest temperature
+    2. max lowest temperature
+    3. max humidity
+    Args:
+        year: a string which contains a year e.g '2006', '2007'
+        path: a string which contains path to a directory containing weather files
+    Returns:
+           a 2d list or int: a 3x2 list maximums where
+                            maximums[0] = [max_highest_temperature, date]
+                            maximums[1] = [max_lowest_temperature, date]
+                            maximums[2] = [max_humidity, date]
+                            date here is a string like: '2002-4-1', '2004-3-14'
+                            or an int if there is an error or failure
+    """
+    # max_temperature contains highest temperature and
+    # date on which the temperature was highest
+    # min_temperature contains lowest temperature and date
+    # max_humidity contains most humidity and date
+    max_temperature = [0, ""]
+    min_temperature = [0, ""]
+    max_humidity = [0, ""]
+    initialized = False
+    # indexes to compare values with.
+    # highest temperature is stored at index 1 of weather reading
+    # lowest temperature is stored at index 3 of weather reading
+    # max humidity is stored at index 7 of weather reading
+    # see the weather files for further explanation
+    indexes = [1, 3, 7]
+    for lines in read_data(year, path):
+        for line in lines:
+            # since readlines() returns ['linedata1\n','linedata2\n',...]
+            # line contains ['linedata\n'] with split by '\n' to get ['linedata','\n]
+            # then we get 'linedata' by grabbing the [0]
+            # lastly,  we split by ',' to get the entries parsed
+            parsed_line = line.split("\n")[0].split(",")
+            date = parsed_line[0]
+            if not initialized:
+                max_temperature = [int(parsed_line[indexes[0]]), date]
+                min_temperature = [int(parsed_line[indexes[1]]), date]
+                max_humidity = [int(parsed_line[indexes[2]]), date]
+                initialized = True
+                continue
+
+            max_temperature_line = get_highest_temperature(parsed_line)
+            if max_temperature_line != -1000:
```

Comment:
instead of checking for `-1000` this would be a check based on `not None:`.

Reply:
done